library/core/src/iter/traits/iterator.rs

```diff
@@ -536,6 +536,28 @@ pub trait Iterator {
         Zip::new(self, other.into_iter())
     }
 
+    /// Places a copy of `separator` between all elements.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// #![feature(iter_intersperse)]
+    ///
+    /// let hello = ["Hello", "World"].iter().copied().intersperse(" ").collect::<String>();
+    /// assert_eq!(hello, "Hello World");
```

Comment:
Add different example and some humor (not tested)
```suggestion
    /// let s = ["996", "icu"].iter().copied().intersperse(".").collect::<String>();
    /// assert_eq!(s, "996.icu");
    ///
    /// let v = [&[9, 9, 6], &[6, 6, 9], &[1, 1, 2]];
    /// let v = v.iter().copied().intersperse(&[0]).collect::<Vec<_>>();
    /// assert_eq!(v, &[9, 9, 6, 0, 6, 6, 9, 0, 1, 1, 2]);
```

Reply:
I'd rather not

Reply:
Haha, how about the second example for `&[u8]`? At least we don't just show `&str`.

Reply:
Another example where `separator` is yielded more than once would be nice. And preferably with slices instead of strings.